### PR TITLE
Syncing with siddhi 5.0.0

### PR DESCRIPTION
**Pull request by sujanan**
## Purpose
$Subject

## Security checks
Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
Ran FindSecurityBugs plugin and verified report? yes
Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java [hunk 1]
```diff
@@ -221,46 +235,75 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
...
-            dataPointsArray = (LinkedList<DataPoint>) map.get("untrainedData");
-            kMeansModel = (KMeansModel) map.get("kMeansModel");
+    public void stop() {
+    }
+
+    static class ExtensionState extends State {
+
+        private static final String KEY_UNTRAINED_DATA = "untrainedData";
+        private static final String KEY_K_MEANS_MODEL = "kMeansModel";
+
```
Comment (mohanvive) [inline component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java]: Don't leave unnecessary line spaces like above. Fix other places as well.

File: component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansMiniBatchSPExtension.java [hunk 2]
```diff
@@ -245,87 +311,76 @@
...
+        private static final String KEY_NUMBER_OF_EVENTS_RECEIVED = "numberOfEventsReceived";
 
-            }
+        private final Map<String, Object> state;
+
+        private KMeansModel kMeansModel;
+
+        private LinkedList<DataPoint> dataPoints;
+
+        private int numberOfEventsReceived;
```
Comment (mohanvive) [inline component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansMiniBatchSPExtension.java:345]: Remove unnecessary line spaces..

**Review by mohanvive — CHANGES_REQUESTED** [review]

File: component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java (hunk 1 above)
Comment (sujanan) [reply component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java]: Fix with 153579a8fccc6c02c19f61500781d59db42180fa

File: component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansMiniBatchSPExtension.java (hunk 2 above)
Comment (sujanan) [reply component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansMiniBatchSPExtension.java:345]: Fix with Fix with 153579a8fccc6c02c19f61500781d59db42180fa

**Review by mohanvive — APPROVED** [review]

**Pull request merged**